File: csharp/AppEncryption/AppEncryption.IntegrationTests/Regression/DecryptMetastoreInteractions.cs
using GoDaddy.Asherah.AppEncryption.IntegrationTests.TestHelpers;

namespace GoDaddy.Asherah.AppEncryption.IntegrationTests.Regression
{
    public class DecryptMetastoreInteractions
    {
        private readonly KeyState cacheIK;
        private readonly KeyState cacheSK;

        public DecryptMetastoreInteractions(KeyState cacheIK, KeyState cacheSK)
        {
            this.cacheIK = cacheIK;
            this.cacheSK = cacheSK;
        }

        protected internal bool ShouldLoadIK()
        {
            return cacheIK == KeyState.Empty;
        }

        protected internal bool ShouldLoadSK()
        {
            if (ShouldLoadIK())
            {
                return cacheSK == KeyState.Empty;
            }

            return false;
        }
    }
}

File: java/app-encryption/src/it/java/com/godaddy/asherah/regression/DecryptMetastoreInteractions.java
package com.godaddy.asherah.regression;

import com.godaddy.asherah.testhelpers.KeyState;

class DecryptMetastoreInteractions {

  private KeyState cacheIK;
  private KeyState cacheSK;

  public DecryptMetastoreInteractions(final KeyState cacheIK, final KeyState cacheSK) {
    this.cacheIK = cacheIK;
    this.cacheSK = cacheSK;
  }

  protected boolean shouldLoadIK() {
    return cacheIK == KeyState.EMPTY;
  }

  protected boolean shouldLoadSK() {
    if (shouldLoadIK()) {
      return cacheSK == KeyState.EMPTY;
    }
    return false;
  }
}

